src/rating/emoticon-rating.js

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+
+// @flow
+
+import React from 'react';
+import type {EmoticonRatingPropsT, RatingStateT} from './types';
+import {StyledRoot, StyledEmoticon} from './styled-components';
+import {getOverrides} from '../helpers/overrides';
+import {ENTER_KEY_CODE, SPACE_KEY_CODE} from './utils';
+
+class EmoticonRating extends React.Component<
+  EmoticonRatingPropsT,
+  RatingStateT,
+> {
+  static defaultProps = {
+    overrides: {},
+  };
+
+  state = {};
+
+  selectItem = (value: number) => {
+    const {onChange} = this.props;
+
+    onChange && onChange({value});
+    this.setState({previewIndex: undefined});
+  };
+
+  updatePreview = (previewIndex?: number) => {
+    this.setState({previewIndex});
+  };
+
+  renderRatingContents = () => {
+    const {overrides = {}, value = -1} = this.props;
+    const {previewIndex} = this.state;
+
+    const [Emoticon, emoticonProps] = getOverrides(
+      overrides.Item,
+      StyledEmoticon,
+    );
+
+    const ratings = [];
+
+    for (let x = 1; x <= 5; x++) {
+      ratings.push(
+        <Emoticon
+          key={x}
+          role="radio"
+          tabIndex={0}
+          aria-setsize={5}
+          aria-checked={x === value}
+          aria-posinset={x}
+          $index={x}
+          $isActive={
+            previewIndex !== undefined ? x === previewIndex : x === value
```

Comment:
The other option could be introducing a `$isPseudoActive` props for elements with a `key` value less than `previewIndex`. This way we could export a single `Rating` component for both `star` and `emoticons` types and only style pseudo-active `StyledStar` items but not `StyledEmoticon`.